//
// Copyright 2019 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package guide.asylo;

import "asylo/enclave.proto";

// A custom message to pass in and out of our enclave.
message Demo {
  // This string value is used for both user input and enclave output.
  optional string value = 1;
}

// The EnclaveInput message that is passed to the enclave can be extended with
// a Demo message to communicate a value our enclave knows to expect.
extend .asylo.EnclaveInput {
  optional Demo quickstart_input = 9001;
}

// The EnclaveOutput message that is passed out of the enclave can be extended
// with a Demo message to communicate a value our driver knows to expect.
extend .asylo.EnclaveOutput {
  optional Demo quickstart_output = 9001;
}
